app.rb

```diff
@@ -32,23 +37,60 @@ def create_person
     name = gets.chomp
     puts 'Enter person age:'
     age = gets.chomp.to_i
+
     if person_type == 'student'
       puts 'Enter classroom:'
-      class_name = gets.chomp
-      puts 'Student has parent permission :'
-      permission = gets.chomp
-      student = Student.new(@people.length + 1, name, age, classroom: class_name, parent_permission: permission)
-      @people << student
+      classroom = gets.chomp
+      puts 'Does the student have parent permission (true/false):'
+      parent_permission = gets.chomp
+
+      student = Student.new(@people.length + 1, name, age, classroom: classroom, parent_permission: parent_permission)
+
+      # Load existing people data from the JSON file
+      people_data = JSON.parse(File.read('./data/people.json'))
+
+      person_data = {
+        'id' => student.id,
+        'name' => student.name,
+        'age' => student.age,
+        'classroom' => student.classroom,
+        'parent_permission' => student.parent_permission,
+        'rentals' => student.rentals
+      }
+
+      # Add the new person object to the existing array
+      people_data << person_data
+
+      # Write the updated array back to the JSON file
+      File.write('./data/people.json', JSON.generate(people_data))
+
```

Comment:
- [ ] Great work refactoring the code to implement suggestions from the previous reviewer. I noticed that the problem persists. When the JSON files are deleted, and I try to create people, books or rental data, I get these errors shown below:

![image](https://github.com/ridabensalem/School-library/assets/77106962/770ecc37-6bb5-428a-82e4-3786ac2c758a)

![image](https://github.com/ridabensalem/School-library/assets/77106962/4094e198-bc92-4c15-a199-4747cc8b8d4a)

The logic is to create the files if they don't exist. 👍 💻 

#### Hint: In the methods for creating the data, you should create a file if it doesn't exist:

```ruby
 File.new('./data/people.json', 'w') unless File.exist?('./data/people.json')
```
#### When reading the files, you could create a safety like this:

```ruby
people_data = File.read('./data/people.json').empty? ? [] : JSON.parse(File.read('./data/people.json'))
```

#### Please ensure that the app functions properly (saving data in JSON files and loading data from JSON files) before submitting it for another review. 👍 💻